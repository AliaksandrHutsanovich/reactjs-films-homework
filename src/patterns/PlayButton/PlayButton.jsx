import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { PlayCircleFilled } from '@ant-design/icons';

import '!style-loader!css-loader!sass-loader!antd/dist/antd.css'; //eslint-disable-line

import { Text } from '../../components';

import styles from './PlayButton.scss';

const PlayButton = ({ onClick }) => (
  <div className={styles.wrapper}>
    <PlayCircleFilled onClick={onClick} className={styles.wrapper__icon} />
    <Text type="white" size={16}>Watch Now</Text>
  </div>
);

PlayButton.propTypes = {
  onClick: PropTypes.func.isRequired,
};

export default memo(PlayButton);
