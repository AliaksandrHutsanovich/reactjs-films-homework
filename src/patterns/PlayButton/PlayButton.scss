.wrapper {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;

  &>:nth-child(n) {
    margin: 10px 0;
  }

  &__icon {
    font-size: 70px;
    cursor: pointer;
    color: rgb(59, 180, 224);
  }
}
