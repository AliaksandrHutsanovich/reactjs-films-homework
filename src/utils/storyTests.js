const storyTests = (stories, renderer) => {
  Object.keys(stories).forEach((storyKey) => {
    if (typeof stories[storyKey] !== 'function') {
      return;
    }
    it(`${storyKey} should be rendered properly`, () => {
      renderer.render(stories[storyKey]());
      const result = renderer.getRenderOutput();

      expect(result).toMatchSnapshot();
    });
  });
};

export default storyTests;
