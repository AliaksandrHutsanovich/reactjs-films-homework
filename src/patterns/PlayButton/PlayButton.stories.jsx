import React from 'react';
import { action } from '@storybook/addon-actions';
import PlayButton from './PlayButton';

export default {
  component: PlayButton,
  title: 'patterns/PlayButton',
  decorators: [
    (story) => (
      <div
        style={{
          width: '400px',
          height: '300px',
          backgroundColor: 'rgba(0, 0, 0, 0.5)',
        }}
      >
        {story()}
      </div>
    ),
  ],
};

export const playButton = () => (
  <PlayButton onClick={action('the play button is clicked')} />
);
