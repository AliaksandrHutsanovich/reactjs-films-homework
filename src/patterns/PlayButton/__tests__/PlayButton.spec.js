import ShallowRenderer from 'react-test-renderer/shallow';
import storyTests from '../../../utils/storyTests';
import * as stories from '../PlayButton.stories';

describe('<PlayButton />', () => {
  const renderer = new ShallowRenderer();

  storyTests(stories, renderer);
});
