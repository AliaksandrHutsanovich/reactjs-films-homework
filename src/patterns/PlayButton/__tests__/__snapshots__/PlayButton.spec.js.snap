// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayButton /> playButton should be rendered properly 1`] = `
<div
  className="wrapper"
>
  <ForwardRef(PlayCircleFilled)
    className="wrapper__icon"
    onClick={[Function]}
  />
  <UNDEFINED
    size={16}
    type="white"
  >
    Watch Now
  </UNDEFINED>
</div>
`;
